### PR TITLE
feat(sdk-coin-algo): refactor Algorand to its own module

### PR DESCRIPTION
**Pull request by bitgoAaron**
* refactors algo specific code from account-lib & bitgo into dedicated module `@bitgo/sdk-coin-algo`
* re-exports code to be backwards compatible
* provides simpler mechanism for registering tokens with CoinFactory

### CODE REVIEW

**Comment by twtaylor** [comment]
image

File: modules/sdk-coin-algo/test/unit/lib/transactionBuilder/assetTransferBuilder.ts [hunk 1]
```diff
@@ -2,9 +2,9 @@ import { coins } from '@bitgo/statics';
 import assert from 'assert';
 import should from 'should';
 import { test } from 'mocha';
```
Comment (ducnguyen-bitgo) [inline modules/sdk-coin-algo/test/unit/lib/transactionBuilder/assetTransferBuilder.ts:4]: We might need to add mocha into our devDependencies I think.

**Comment by ducnguyen-bitgo** [comment]
Just one minor change in devDependencies for adding `mocha` and we should be good to go.

File: modules/sdk-coin-algo/test/unit/lib/transactionBuilder/assetTransferBuilder.ts (hunk 1 above)
Comment (bitgoAaron) [reply modules/sdk-coin-algo/test/unit/lib/transactionBuilder/assetTransferBuilder.ts:4]: actually `mocha` needs to only be in the root package. We ran into issues with it being included in multiple places when refactoring TestBitgo.

Comment (ducnguyen-bitgo) [reply modules/sdk-coin-algo/test/unit/lib/transactionBuilder/assetTransferBuilder.ts:4]: I see. Then everything looks good to me! Approving it right now.

**Review by ducnguyen-bitgo — APPROVED** [review]

**Pull request merged**